libs/flags/json_flag.go

```diff
@@ -33,7 +37,34 @@ func (j *JsonFlag) Unmarshal(v any) error {
 	if j.raw == nil {
 		return nil
 	}
-	return json.Unmarshal(j.raw, v)
+
+	dv, err := jsonloader.LoadJSON(j.raw)
+	if err != nil {
+		return err
+	}
+
+	// First normalize the input data.
+	// It will convert all the values to the correct types.
+	// For example string lterals for booleans and integers will be converted to the correct types.
+	nv, diags := convert.Normalize(v, dv)
+	if len(diags) > 0 {
+		summary := ""
+		for _, diag := range diags {
+			summary += fmt.Sprintf("- %s\n", diag.Summary)
+		}
+		return fmt.Errorf("json input error:\n%v", summary)
```

Comment:
Could we return the diags instead? If we do, then we can leverage the same renderer and include the location information in the warnings/errors.

If we merge this as is, there will be a class of invocations that worked before (for ex some unknown key in the input) that will now fail. You could argue this is fine, but there could be inputs that have superfluous fields (known to be ignored) for other purposes.